music_assistant/server/providers/snapcast/__init__.py

```diff
@@ -0,0 +1,305 @@
+    await prov.handle_setup()
+    return prov
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+    return (
+        ConfigEntry(
+            key=CONF_SNAPCAST_SERVER_HOST,
+            type=ConfigEntryType.STRING,
+            default_value="127.0.0.1",
+            label="Snapcast server ip",
+            required=True,
+        ),
+        ConfigEntry(
+            key=CONF_SNAPCAST_SERVER_CONTROL_PORT,
+            type=ConfigEntryType.INTEGER,
+            default_value="1705",
+            label="Snapcast control port",
+            required=True,
+        ),
+    )
+
+
+class SnapCastProvider(PlayerProvider):
+    """Player provider for Snapcast based players."""
+
+    _snapserver: [asyncio.Server | asyncio.BaseTransport]
+    snapcast_server_host: str
+    snapcast_server_control_port: int
+
+    async def handle_setup(self) -> None:
+        """Handle async initialization of the provider."""
+        self.snapcast_server_host = self.config.get_value(CONF_SNAPCAST_SERVER_HOST)
+        self.snapcast_server_control_port = self.config.get_value(CONF_SNAPCAST_SERVER_CONTROL_PORT)
+        try:
+            self._snapserver = await snapcast.control.create_server(
+                self.mass.loop,
+                self.snapcast_server_host,
+                port=self.snapcast_server_control_port,
+                reconnect=True,
+            )
+            self._snapserver.set_on_update_callback(self._handle_update)
+            self._handle_update()
+            self.logger.info(
+                f"Started Snapserver connection on:"
+                f"{self.snapcast_server_host}:{self.snapcast_server_control_port}"
+            )
+        except OSError:
+            raise SetupFailedError("Unable to start the Snapserver connection ?")
+
+    def _handle_update(self):
+        for snap_client in self._snapserver.clients:
+            self._handle_player_update(snap_client)
+            snap_client.set_callback(self._handle_player_update)
+
+    def _handle_player_update(self, snap_client):
+        player_id = snap_client.identifier
+        player = self.mass.players.get(player_id, raise_unavailable=False)
+        if not player:
+            player = Player(
+                player_id=player_id,
+                provider=self.domain,
+                type=PlayerType.PLAYER,
+                name=snap_client.friendly_name,
+                available=True,
+                powered=snap_client.connected,
+                device_info=DeviceInfo(),
```

Comment:
Maybe provide some default name here like SnapCast Player ?

Reply:
I can provide more specific information about the device

Reply:
OK, please do so in a follow up PR then.

Reply:
This is what I found the most accurate. 

```
 device_info=DeviceInfo(
                    model=snap_client._client.get("host").get("os"),
                    address=snap_client._client.get("host").get("ip"),
                    manufacturer=snap_client._client.get("host").get("arch"),
                ),
```